# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/find/nsFind.h
# Commit: e07e5ea2f75e
# Full Hash: e07e5ea2f75e971967decbbe8f42aff80987fda2
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-02-16 08:52:20
# Regressor Bug: 1945862
# File Overlap Count: 1
# Description:
#   Bug 1945862 - Make `nsFind` respect the end point correctly. r=emilio
#   
#   This patch refactors the find functionality in a few ways:
#   - A C++-only method `FindFromRangeBoundaries()` is introduced,
#     which uses `RangeBoundary`s instead of collapsed `nsRange`s
# ==============================================================================

diff -r 5a1f8c0bb78f -r e07e5ea2f75e toolkit/components/find/nsFind.h
--- a/toolkit/components/find/nsFind.h	Sat Feb 15 15:19:21 2025 +0000
+++ b/toolkit/components/find/nsFind.h	Sat Feb 15 16:02:34 2025 +0000
@@ -13,15 +13,12 @@
 #include "nsCycleCollectionParticipant.h"
 #include "nsINode.h"
 #include "mozilla/intl/Segmenter.h"
+#include "mozilla/RangeBoundary.h"
 
 #define NS_FIND_CONTRACTID "@mozilla.org/embedcomp/rangefind;1"
 
-#define NS_FIND_CID                                  \
-  {                                                  \
-    0x471f4944, 0x1dd2, 0x11b2, {                    \
-      0x87, 0xac, 0x90, 0xbe, 0x0a, 0x51, 0xd6, 0x09 \
-    }                                                \
-  }
+#define NS_FIND_CID \
+  {0x471f4944, 0x1dd2, 0x11b2, {0x87, 0xac, 0x90, 0xbe, 0x0a, 0x51, 0xd6, 0x09}}
 
 class nsFind : public nsIFind {
  public:
@@ -42,6 +39,14 @@
     mWordEndBounded = aWordEndBounded;
   }
 
+  void SetNodeIndexCache(nsContentUtils::NodeIndexCache* aCache) {
+    mNodeIndexCache = aCache;
+  }
+
+  already_AddRefed<nsRange> FindFromRangeBoundaries(
+      const nsAString& aPatText, const mozilla::RangeBoundary& aStartPoint,
+      const mozilla::RangeBoundary& aEndPoint);
+
  protected:
   virtual ~nsFind() = default;
 
@@ -53,6 +58,7 @@
   bool mWordStartBounded = false;
   bool mWordEndBounded = false;
   mozilla::intl::WordBreakIteratorUtf16 mWordBreakIter{nullptr};
+  nsContentUtils::NodeIndexCache* mNodeIndexCache = nullptr;
   struct State;
   class StateRestorer;
 